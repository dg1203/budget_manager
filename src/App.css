body {
  margin: 0;
}
a {
  text-decoration: none;
  color: white;
}
form {
  max-width: 700px;
  width: 100%;
}