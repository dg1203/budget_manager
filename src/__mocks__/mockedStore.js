import { createStore } from "redux";
import rootReducer from "../reducers/rootReducer";

const initialState = {
  jars: [
    {
      id: "dasas",
      currency: "",
      name: "Test",
      description: "Test",
      resources: []
    }
  ],
  logs: [],
  defaultJar: "fds",
  message: null
};

const store = createStore(rootReducer, initialState);

export default store;
