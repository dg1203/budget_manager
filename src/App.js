import React from "react";
import "./App.css";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Container } from "@material-ui/core";

import Header from "./components/Header";
import Home from "./pages/Home";
import AddJar from "./pages/AddJar";

function App() {
  return (
    <div className="App">
      <Router>
        <Header />
        <Container maxWidth="lg" style={{ padding: "50px 15px" }}>
          <Switch>
            <Route path="/" component={Home} exact />
            <Route path="/jar/add" component={AddJar} exact />
          </Switch>
        </Container>
      </Router>
    </div>
  );
}

export default App;
