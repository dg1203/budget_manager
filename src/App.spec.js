import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import App from "./App";
import mockedStore from "./__mocks__/mockedStore";

const renderComponent = (store = mockedStore) =>
  render(
    <Provider store={mockedStore}>
      <App />
    </Provider>
  );

describe("App component", () => {
  it("renders the component", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
