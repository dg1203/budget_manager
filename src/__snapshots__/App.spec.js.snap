// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <button
              aria-label="menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
              data-testid="open-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <a
              href="/"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                Zarządzanie budżetem
              </h6>
            </a>
          </div>
        </header>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
          style="padding: 50px 15px;"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <div
                class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-margin-1 MuiTypography-h5"
                  >
                    Test
                  </h2>
                  <p
                    class="MuiTypography-root makeStyles-margin-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
                  >
                    Test
                  </p>
                  <p
                    class="MuiTypography-root makeStyles-margin-1 MuiTypography-body2"
                  >
                    Elementów w słoiku: 
                    0
                    <br />
                    
                  </p>
                  <a
                    class="makeStyles-margin-1"
                    href="/jar/edit/dasas"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Więcej
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            data-testid="open-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <a
            href="/"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              Zarządzanie budżetem
            </h6>
          </a>
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
        style="padding: 50px 15px;"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <h2
                  class="MuiTypography-root makeStyles-margin-1 MuiTypography-h5"
                >
                  Test
                </h2>
                <p
                  class="MuiTypography-root makeStyles-margin-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  Test
                </p>
                <p
                  class="MuiTypography-root makeStyles-margin-1 MuiTypography-body2"
                >
                  Elementów w słoiku: 
                  0
                  <br />
                  
                </p>
                <a
                  class="makeStyles-margin-1"
                  href="/jar/edit/dasas"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Więcej
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
